client/src/reducers/categoricalSelection.js

```diff
@@ -30,30 +32,25 @@ const CategoricalSelection = (
       const { world } = nextSharedState;
       const names = CH.selectableCategoryNames(
         world.schema,
-        CH.maxCategoryItems(prevSharedState.config),
         dataframe.colIndex.labels()
       );
       if (names.length === 0) return state;
       return {
         ...state,
-        ...CH.createCategoricalSelection(world, names),
+        ...CH.createCategoricalSelection(names),
       };
     }
 
     case "categorical metadata filter select": {
       /*
       Set the specific category in this field to false
       */
-      const newCategoryValueSelected = Array.from(
-        state[action.metadataField].categoryValueSelected
-      );
-      newCategoryValueSelected[action.categoryIndex] = true;
+      const { metadataField, label } = action;
+      const newSelected = new Map(state[metadataField]);
```

Comment:
I personally love `Map`, but `Map` is not serializable, so if ever want to serialize the Redux store as JSON, extra processing is needed. If that's not a concern, then we're good!

>ES2015 Map is not a serializable data structure so you need extra works if you need to persist your Store data. Redux DevTools doesn’t display data of ES2015 Map by default, so you have to specify serialize option for this.

[source](https://medium.com/@koba04/redux-store-patterns-using-es2015-map-f62804755675#4955)

Reply:
we can't use an object, as the keys are native types (and must be native types).  I used Map explicitly, not as a replacement for Object.

Reply:
Ahh I see, so `state[metadataField]` returns `Map`, and we create a copy of it via `new Map(state[metadataField])` 👌 !

Reply:
yep!